### PR TITLE
9307 remove hardcoded cerner api

### PR DESCRIPTION
**Pull request by apisandipas**
## Summary
Removes any mention of `CERNER_FACILITY_IDS` as this is now managed in Drupal, the code was now obsolete. Removes DSOT aliasing as now DSOT is the only source for selectors.

## Related issue(s)
- Issue ticket: department-of-veterans-affairs/va.gov-cms/issues/9307
-  Relates to : department-of-veterans-affairs/va.gov-cms#9078

## Testing done
Manual testing of all 5 Cerner-related widgets in the unauthenticated, authenticated MHV and authenticated Cerner scenarios.
- [x] [Refill and track your prescriptions](https://www.va.gov/health-care/refill-track-prescriptions)
- [x] [Send a secure message to your health care team](https://www.va.gov/health-care/secure-messaging)
- [x] [Schedule and manage health appointments](https://www.va.gov/health-care/schedule-view-va-appointments)
- [x] [View your lab and test results](https://www.va.gov/health-care/view-test-and-lab-results)
- [x] [Apply now for VA health care](https://www.va.gov/health-care/apply/application)

<details>
  <summary>Unauthenticated widgets</summary>
Screenshot 2023-08-24 at 3 19 08 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 18 59 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 18 50 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 18 42 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 18 33 PM
</details>

<details>
  <summary>Authenticated MHV widgets</summary>
Screenshot 2023-08-24 at 3 17 56 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 17 47 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 17 21 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 17 10 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 16 54 PM

</details>

<details>
  <summary>Authenticated Cerner widgets</summary>
Screenshot 2023-08-24 at 3 15 53 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 15 18 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 14 45 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 14 01 PM
<br/>
<br/>
Screenshot 2023-08-24 at 3 11 51 PM
</details>
 
## What areas of the site does it impact?
This removal affects all part of the app that previously relied on the `CERNER_FACILITY_IDS` constant, which has been deprecated and now removed. That is to say, it should have no effect at this point.

## Acceptance criteria
- [x] No code left over from hard-code/synchronous method
- [x] I updated unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x] Unit tests passing
- [x] Build completes

### CODE REVIEW

File: src/platform/user/selectors.js
Comment (ryguyk) [inline src/platform/user/selectors.js:26]: This entire selector should be removed. It won't work correctly without the hard-coded data, and it should not be called anywhere. The same is true for:
`selectIsCernerOnlyPatient`
`selectIsCernerPatient`
`selectCernerRxFacilities`
`selectCernerMessagingFacilities`
`selectCernerAppointmentsFacilities`
`selectCernerMedicalRecordsFacilities`
`selectCernerTestResultsFacilities`

HOWEVER, it looks like consuming apps are still using at least one of these. We probably should not commit changes that remove these selectors until we are certain it is okay to do so.

**Review by ryguyk — CHANGES_REQUESTED** [review]
This is a great start, but we should chat about it. There is some more code that can be removed, but timing is important.

**Comment by jilladams** [comment]
Asked @ryguyk for some help to resolve conflicts here for merge. He rebased the branch on `main` which required a force push to GitHub. The next person who picks this back up will need to pull down a fresh copy of the branch before doing anything. Ryan also helped identify some outstanding problems. 

In this PR, we are removing the `selectPatientFacilities` function, which was built on `CERNER_FACILITY_IDS`. However: there are references to `selectPatientFacilities` elsewhere in the code that are preventing tests from passing, and references in functional product code that belong to other teams so the build is failing.  So: we still have mentions to mitigate before we could merge this PR. (FYI @wesrowe )

## **Remaining code references** to review / fix

### Other teams
I'll follow up on these
1. **Health products**
codeowner: `@department-of-veterans-affairs/va-cto-health-products`
file: `src/applications/coronavirus-vaccination/components/SelectFacilityWidget/index.jsx`

2. **Personalization**
codeowner: `@department-of-veterans-affairs/vsa-authd-exp-frontend @department-of-veterans-affairs/benefits-team-1-frontend` (personalization/profile)
files: 
`src/applications/personalization/profile/components/notification-settings/NotificationChannel.jsx`
`src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx`

3. **VAOS**
codeowner: vaos file
file: `src/applications/vaos/components/EnrolledRoute.jsx`

### Public Websites purview
4. **Unit tests** for the old selector: This selector has been removed, so the references are failing during build:
`src/platform/user/tests/selectors.unit.spec.js`- PW should check for and remove any tests that reference selectors that have been removed here: https://github.com/department-of-veterans-affairs/vets-website/pull/23036/files#diff-9ec1d9d6f4976449cb9b5f337f1946801ecb0732d3f216d922ab2f790db5f073L27

5. References in a few different exportsFile.js files to a number of things that have been removed:
`src/platform/user/exportsFile.js` - PW should make this change and we could do in this PR. We are not codeowners of the file.
`src/platform/utilities/exportsFile.js` - same
We need to check for any references to: 
```
  CERNER_APPOINTMENTS_BLOCKLIST,
  CERNER_FACILITY_IDS,
  CERNER_MEDICAL_RECORDS_BLOCKLIST,
  CERNER_MESSAGING_BLOCKLIST,
  CERNER_RX_BLOCKLIST,
  CERNER_TEST_RESULTS_BLOCKLIST,
```

File: src/applications/static-pages/health-care-manage-benefits/refill-track-prescriptions-page/components/App/index.js
Comment (jilladams) [inline src/applications/static-pages/health-care-manage-benefits/refill-track-prescriptions-page/components/App/index.js:51]: @ryguyk before I finish reaching out to teams about this, can you help me understand one more thing?  I had thought we said: we are removing `selectPatientFacilities` so should have no references to it anywhere in the code anymore. However: in this change, we used to have `selectPatientFacilitiesDsot` function, now being renamed here to `selectPatientFacilities`.  Are these references of our own also now wrong? 

I'd like to make sure I'm clear for other teams who reference `selectPatientFacilities` - what is it I am actually asking them to do?

Comment (ryguyk) [reply src/applications/static-pages/health-care-manage-benefits/refill-track-prescriptions-page/components/App/index.js:51]: It's not that `selectPatientFacilities` should never appear in our codebase. It will. The new selector and the old selector are _both_ named that. The real point here is to ensure that there is no code trying to import (and call) the _old_ one.

In the example above, the call to `selectPatientFacilities(state)` on line 51 is a call to the _new_ selector, which is imported on line 7 from the `cerner-dsot` path. 

As an extra point of clarity, this diff highlights that line 7 is slightly changed. It was already importing the new selector (which, again, is named `selectPatientFacilities` just like the old one), but it was previously creating an alias `selectPatientFacilitiesDsot` so that it could be differentiated against the old `selectPatientFacilities` when we were conditionally calling each of them based on our Flipper.

**Comment by jilladams** [comment]
Update! We might be unblocked.
https://github.com/department-of-veterans-affairs/va.gov-team/issues/49612 is still open, but Patrick Bateman shipped this change: https://github.com/department-of-veterans-affairs/vets-website/pull/24354, which might remove the last barrier to deprecating the old data method. Need to verify, and if that's true, may be able to finalize /merge.

**Comment by jilladams** [comment]
Adding @randimays and @dsasser for general awareness of the topics, not because code review here is required.

**Comment by jilladams** [comment]
Per https://dsva.slack.com/archives/CMNQT72LX/p1691421830026429?thread_ts=1673026289.887489&cid=CMNQT72LX, 
https://github.com/department-of-veterans-affairs/vets-website/pull/25068 includes the remaining change from Appointments team and should unblock us.

**Review by radavis — APPROVED** [review]

File: src/platform/user/selectors.js
Comment (randimays) [reply src/platform/user/selectors.js]: Consuming apps removed all usage except one which is covered in this PR. Done 👍

**Comment by radavis** [comment]
It would be excellent if we could get some documentation on the `selectPatientFacilities` selector to confirm `selectDrupalStaticData(state).vamcEhrData.loading` has completed before accessing this selector.

I'm unsure if this is the best place to address this concern. Just tell me to open a support ticket with Platform, if necessary. :)

**Comment by randimays** [comment]
@ryguyk Could you please re-review this PR when you have some bandwidth? If you have some additional context for @radavis's [question above](https://github.com/department-of-veterans-affairs/vets-website/pull/23036#issuecomment-1693538037) that would be most helpful.

**Comment by randimays** [comment]
@department-of-veterans-affairs/va-cto-health-products @department-of-veterans-affairs/vfs-vaos-fe @department-of-veterans-affairs/va-platform-cop-frontend @chriskim2311 @dsasser 

Will you please review this PR when you have some bandwidth? Thank you!

**Review by acrollet — CHANGES_REQUESTED** [review]
Could the various imports be updated to use the non-deprecated syntax? thanks 🙂

**Review by vbahinwillit — APPROVED** [review]
Changes for appointments LGTM

**Review by powellkerry — APPROVED** [review]

**Review by acrollet — APPROVED** [review]
Changes seem reasonable. @radavis have your concerns been addressed?

**Review by chriskim2311 — APPROVED** [review]
LGTM

**Comment by ryguyk** [comment]
> It would be excellent if we could get some documentation on the `selectPatientFacilities` selector to confirm `selectDrupalStaticData(state).vamcEhrData.loading` is false and fetching has completed before accessing this selector.
> 
> I'm unsure if this is the best place to address this concern. Just tell me to open a support ticket with Platform, if necessary. :)

@radavis I'm not sure I understand exactly what you're looking for. Are you wondering if there is documentation that addresses your question? Or you'd like to add documentation somewhere? Or you just want that question answered?

If I correctly understand what you're poking at, it shouldn't matter if the selector `selectPatientFcilities` is called before the data is finished loading. This is the magic of Redux and React working together. If your component is properly connected to the Redux store, it should be re-rendered whenever the fetch completes and the Redux store is updated to reflect that.

**Comment by radavis** [comment]
@ryguyk, If I have logic that redirects users based on the results of `selectPatientFacilities` and `selectDrupalStaticData(state).vamcEhrData.loading` is true, what will happen?

Here's a writeup on a bug I experienced around this: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/engineering/61496-mhvlp-redirect-bugfix.md

I'm requesting we add jsdoc comments on the selector function to tell others that this data is not valid until `selectDrupalStaticData(state).vamcEhrData.loading` is false and `selectDrupalStaticData(state).vamcEhrData.errors` is null.

**Comment by randimays** [comment]
@radavis @ryguyk I think the loading handling is out-of-scope for this ticket. @ryguyk, would this be something Facilities handles or Public Websites?

**Comment by radavis** [comment]
This is all I'm requesting: 

```
/**
 * Augments state.user.profile.facilities with data from vamc-ehr.json.
 * Valid state depends on selectDrupalStaticData(state).vamcEhrData.loading
 * being false.
 * **TODO**: dispatch FETCH_STATIC_DATA_FAILED and set vamcEhrData.errors when
 *  when fetching fails.
 * @param {Object} state
 * @returns {Object} the current user's facilities, including a facility's EHR data.
 */
export const selectPatientFacilities = state =>
// ...
```

https://github.com/department-of-veterans-affairs/vets-website/blob/662f912ba2dda9e4ea2b58fa1273322cb69c4a99/src/platform/user/selectors.js#L27C1-L35C4****

**Comment by randimays** [comment]
@radavis Added

**Comment by jilladams** [comment]
@randimays thus far Public Websites owns all this logic. If we feel there are reasons it could / should sit with Facilities for future work, let's discuss first and I can help sort out ownership with DaveC.

**Review by simiadebowale — APPROVED** [review]
LGTM

**Comment by ryguyk** [comment]
@radavis Thank you for your clarification and input. I didn't understand your use case initially. You bring up a good point. My intention when I initially wrote this was to mimic the old behavior as closely as possible, namely to be able to simply call the selector in the same fashion. It does depend on the data being loaded, but that consideration can be abstracted away when relying on the Redux connection that triggers component refreshes. I hadn't considered the situation where a redirect would be dependent on this data. In that case, timing is certainly an important consideration.

I think this ultimately boils down to clearly communicating that the data is asynchronously fetched. If that is understood, the need to check/wait/etc. feels fairly clear. It's perhaps not clear, though, and so I think your jsdoc addition is quite useful. Thank you!

**Review by ryguyk — APPROVED** [review]
Approved!! This was a long time in the making. Thanks @randimays for getting this over the finish line :)

**Comment by jilladams** [comment]
Original target deadline: December 1, 2022. :lolsob: but better late than never, y'all.

**Pull request merged**
